### PR TITLE
Adjust and move Effective Dart's linter mention

### PR DESCRIPTION
**Pull request by parlough**
Contributes to #1319

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

File: src/_guides/language/effective-dart/index.md [hunk 1]
```diff
@@ -107,6 +95,19 @@ your laces tied correctly. Things aren't that bad. Most of the guidelines here
 are common sense and we're all reasonable people. The goal, as always, is nice,
 readable and maintainable code.
 
+The Dart analyzer has a linter to help you write good, consistent code.
+If a linter rule exists that can help you follow a guideline,
+then the guideline links to that rule using the following format:
```
Comment (atsansone) [inline src/_guides/language/effective-dart/index.md:100]: **suggestion:** Based on the other pages that you reference, I would update the language somewhat.
```suggestion
The Dart analyzer provides a linter to identify potential source code problems.
If your code triggers a [linter rule](/tools/linter-rules),
then a link to that rule appears. The link uses the following format:
```

**question:** Where does the link display? In the IDE? It's a little unclear here.

File: src/_guides/language/effective-dart/index.md [hunk 2]
```diff
@@ -107,6 +95,19 @@ your laces tied correctly. Things aren't that bad. Most of the guidelines here
...
+
+{% include linter-rule-mention.md rule="unnecessary_getters_setters" %}
+
+For help using the linter,
+see [Enabling linter rules][]
+and the list of [linter rules][].
```
Comment (atsansone) [inline src/_guides/language/effective-dart/index.md:106]: **suggestion:** Encourage action in "see also" type passages.
```suggestion
To learn how to use the linter,
see [Enabling linter rules][]
and the list of [linter rules][].
```

**Review by atsansone — APPROVED** [review]
LGTM % 2 suggestions and a question.

File: src/_guides/language/effective-dart/index.md (hunk 1 above)
Comment (parlough) [reply src/_guides/language/effective-dart/index.md:100]: This text is describing the format of the Effective Dart pages, where if that guideline has a cooresponding linter rule, then it links to it. The goal is to help readers understand what that mention is.

Here is an example:
Effective Dart linter rule link

Comment (atsansone) [reply src/_guides/language/effective-dart/index.md:100]: @parlough : Then my changes probably don't apply. Maybe make it more clear that you're writing about the highlighted issue in the documentation, not that it's in the code or IDE.

**Pull request merged**